### ISSUE TITLE
ESI does not return X-Pages header for /markets/{region_id}/orders/

### ISSUE THREAD

**Issue by ivoronin** [ccpgames/esi-issues#474]
# Bug
ESI does not return X-Pages header for /markets/{region_id}/orders/

### Request
https://esi.tech.ccp.is/latest/markets/10000002/orders/?datasource=tranquility&order_type=buy&page=1

### Response
```json
{
  "access-control-allow-credentials": "true",
  "access-control-allow-headers": "Content-Type,Authorization,X-User-Agent",
  "access-control-allow-methods": "GET,OPTIONS",
  "access-control-allow-origin": "*",
  "access-control-expose-headers": "Content-Type,Warning",
  "access-control-max-age": "600",
  "cache-control": "public",
  "content-encoding": "gzip",
  "content-type": "application/json",
  "date": "Thu, 29 Jun 2017 05:47:45 GMT",
  "expires": "Thu, 29 Jun 2017 05:45:42 GMT",
  "last-modified": "Thu, 29 Jun 2017 05:40:42 GMT",
  "strict-transport-security": "max-age=31536000",
  "vary": "Accept-Encoding",
  "via": "1.1 google",
  "alt-svc": "clear",
  "x-firefox-spdy": "h2"
}
```
### Expected
X-Pages header to be present

**Comment by Kyria** [comment]
I got it when I call that page...
```
access-control-allow-credentials:true
access-control-allow-headers:Content-Type,Authorization,X-User-Agent
access-control-allow-methods:GET,OPTIONS
access-control-allow-origin:*
access-control-expose-headers:Content-Type,Warning
access-control-max-age:600
age:23
alt-svc:clear
cache-control:public
content-encoding:gzip
content-length:143969
content-type:application/json
date:Thu, 29 Jun 2017 06:55:15 GMT
expires:Thu, 29 Jun 2017 06:59:49 GMT
last-modified:Thu, 29 Jun 2017 06:54:49 GMT
status:200
strict-transport-security:max-age=31536000
vary:Accept-Encoding
via:1.1 google
x-pages:27
```

**Comment by ivoronin** [comment]
Looks like this issue is intermittent:

# Test 1
curl -v 'https://esi.tech.ccp.is/latest/markets/10000002/orders/?datasource=tranquility&order_type=buy&page=1'  > /dev/null
```
...
< HTTP/1.1 200 OK
< Access-Control-Allow-Credentials: true
< Access-Control-Allow-Headers: Content-Type,Authorization,X-User-Agent
< Access-Control-Allow-Methods: GET,OPTIONS
< Access-Control-Allow-Origin: *
< Access-Control-Expose-Headers: Content-Type,Warning
< Access-Control-Max-Age: 600
< Content-Length: 2384482
< Content-Type: application/json
< Date: Thu, 29 Jun 2017 07:13:56 GMT
< Expires: Thu, 29 Jun 2017 07:18:28 GMT
< Last-Modified: Thu, 29 Jun 2017 07:13:28 GMT
< Strict-Transport-Security: max-age=31536000
< Vary: Accept-Encoding
< Via: 1.1 google
< Age: 175
< Cache-Control: public
< Alt-Svc: clear
...
```

# Test 2 (few seconds later)

curl -v 'https://esi.tech.ccp.is/latest/markets/10000002/orders/?datasource=tranquility&order_type=buy&page=1'  > /dev/null
```
...
< HTTP/1.1 200 OK
< Access-Control-Allow-Credentials: true
< Access-Control-Allow-Headers: Content-Type,Authorization,X-User-Agent
< Access-Control-Allow-Methods: GET,OPTIONS
< Access-Control-Allow-Origin: *
< Access-Control-Expose-Headers: Content-Type,Warning
< Access-Control-Max-Age: 600
< Content-Length: 2384482
< Content-Type: application/json
< Date: Thu, 29 Jun 2017 07:13:36 GMT
< Expires: Thu, 29 Jun 2017 07:18:28 GMT
< Last-Modified: Thu, 29 Jun 2017 07:13:28 GMT
< Strict-Transport-Security: max-age=31536000
< Vary: Accept-Encoding
< X-Pages: 27
< Via: 1.1 google
< Cache-Control: public
< Age: 232
< Alt-Svc: clear
...
```

**Comment by aquarhead** [comment]
It should only return `X-Pages` for when `order_type=all` ...

**Comment by a-tal** [comment]
resolved in ESI 0.5.2
